product_form_sale_link/tests/test_product.py

```diff
@@ -0,0 +1,172 @@
+from odoo.tests import TransactionCase, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestProductSalesCount(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Create test users with different access rights
+        sale_manager_group = cls.env.ref("sales_team.group_sale_manager")
+        sale_user_group = cls.env.ref("sales_team.group_sale_salesman")
+        product_manager_group = cls.env.ref("product.group_product_manager")
+
+        cls.user_salesman = cls.env["res.users"].create(
+            {
+                "name": "Test Salesman",
+                "login": "test_salesman",
+                "email": "test_salesman@example.com",
+                "groups_id": [
+                    (
+                        6,
+                        0,
+                        [
+                            sale_manager_group.id,
+                            sale_user_group.id,
+                            cls.env.ref("base.group_user").id,
+                            cls.env.ref("base.group_partner_manager").id,
+                        ],
+                    )
+                ],
+            }
+        )
+
+        cls.user_employee = cls.env["res.users"].create(
+            {
+                "name": "Test Employee",
+                "login": "test_employee",
+                "email": "test_employee@example.com",
+                "groups_id": [
+                    (
+                        6,
+                        0,
+                        [
+                            cls.env.ref("base.group_user").id,
+                            product_manager_group.id,
+                        ],
+                    )
+                ],
+            }
+        )
+
+        cls.currency = cls.env["res.currency"].create(
```

Comment:
Don't create all this useless stuff. You don't care about the sales orders amounts. You only care about the number.